caikit_nlp/modules/text_embedding/embedding.py

```diff
@@ -0,0 +1,413 @@
+        float_list_list = res.tolist()
+        return SentenceSimilarityResults(
+            results=[SentenceSimilarityScores(fl) for fl in float_list_list],
+            producer_id=self.PRODUCER_ID,
+        )
+
+    @RerankTask.taskmethod()
+    def run_rerank_query(
+        self,
+        query: str,
+        documents: List[JsonDict],
+        top_n: Optional[int] = None,
+        return_documents: bool = True,
+        return_query: bool = True,
+        return_text: bool = True,
+    ) -> RerankResult:
+        """Rerank the documents returning the most relevant top_n in order for this query.
+        Args:
+            query: str
+                Query is the source string to be compared to the text of the documents.
+            documents:  List[JsonDict]
+                Each document is a dict. The text value is used for comparison to the query.
+                If there is no text key, then _text is used and finally default is "".
+            top_n:  Optional[int]
+                Results for the top n most relevant documents will be returned.
+                If top_n is not provided or (not > 0), then all are returned.
+            return_documents:  bool
+                Default True
+                Setting to False will disable returning of the input document (index is returned).
+            return_query:  bool
+                Default True
+                Setting to False will disable returning of the query (results are in query order)
+            return_text:  bool
+                Default True
+                Setting to False will disable returning of document text string that was used.
+        Returns:
+            RerankResult
+                Returns the (top_n) scores in relevance order (most relevant first).
+                The results always include a score and index which may be used to find the document
+                in the original documents list. Optionally, the results also contain the entire
+                document with its score (for use in chaining) and for convenience the query and
+                text used for comparison may be returned.
+
+        """
+
+        error.type_check(
+            "<NLP05323654E>",
+            str,
+            query=query,
+        )
+
+        results = self.run_rerank_queries(
+            queries=[query],
+            documents=documents,
+            top_n=top_n,
+            return_documents=return_documents,
+            return_queries=return_query,
+            return_text=return_text,
+        ).results
+
+        if results:
+            return RerankResult(result=results[0], producer_id=self.PRODUCER_ID)
+
+        RerankResult(
+            producer_id=self.PRODUCER_ID,
+            result=RerankScore(
+                scores=[],
+                query=query if return_query else None,
+            ),
+        )
+
+    @RerankTasks.taskmethod()
+    def run_rerank_queries(
+        self,
+        queries: List[str],
+        documents: List[JsonDict],
```

Comment:
is jsondict and dict same? Wondering if the pure python experience for using these functions will be problematic? i.e if I want to use this function in a notebook or something and then I'll need to first convert my documents to JsonDict ?

Reply:
Caikit won't allow Dict[str, Any].  I would have to provide data objects to wrap Any and that is exactly what JsonDict does for me. So this only works with JsonDict unless there is some smaller piece where I can use dict and hide part of the JsonDict usage, but if that is even allowed we'd still have JsonDict output so I think that is not helpful.

So in general, this must be the caikit way, but if you think I can back-off part of that and use Python please clarify and I'll do some more testing to see where that breaks down.   I'd love to use plain Python.